execute/aggregate.go

```diff
@@ -1,33 +1,159 @@
 package execute
 
 import (
+	"github.com/apache/arrow/go/arrow/memory"
 	"github.com/influxdata/flux"
 	"github.com/influxdata/flux/array"
+	"github.com/influxdata/flux/arrow"
 	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute/table"
 	"github.com/influxdata/flux/internal/errors"
-	"github.com/influxdata/flux/memory"
 	"github.com/influxdata/flux/plan"
 )
 
+// AggregateTransformation implements a transformation that aggregates
+// the results from multiple TableView values and then outputs a Table
+// with the same group key.
+//
+// This is similar to NarrowTransformation that it does not modify the group key,
+// but different because it will only output a table when the key is flushed.
+type AggregateTransformation interface {
+	// Aggregate will process the TableView with the state from the previous
+	// time a table with this group key was invoked.
+	// If this group key has never been invoked before, the
+	// state will be nil.
+	// The transformation should return the new state and a boolean
+	// value of true if the state was created or modified.
+	// If false is returned, the new state will be discarded and any
+	// old state will be kept.
+	// It is ok for the transformation to modify the state if it is
+	// a pointer. This is both allowed and recommended.
+	Aggregate(chunk table.Chunk, state interface{}, mem memory.Allocator) (interface{}, bool, error)
+
+	// Compute will signal the AggregateTransformation to compute
+	// the aggregate for the given key from the provided state.
+	Compute(key flux.GroupKey, state interface{}, d *TransportDataset, mem memory.Allocator) error
+}
+
 type aggregateTransformation struct {
-	ExecutionNode
-	d     Dataset
-	cache TableBuilderCache
-	agg   Aggregate
+	t AggregateTransformation
+	d *TransportDataset
+}
+
+// NewAggregateTransformation constructs a Transformation and Dataset
+// using the aggregateTransformation implementation.
+func NewAggregateTransformation(id DatasetID, t AggregateTransformation, mem memory.Allocator) (Transformation, Dataset, error) {
+	tr := &aggregateTransformation{
+		t: t,
+		d: NewTransportDataset(id, mem),
+	}
+	return tr, tr.d, nil
+}
+
+// ProcessMessage will process the incoming message.
+func (n *aggregateTransformation) ProcessMessage(m Message) error {
```

Comment:
Please let me know what n stands for here. if it's not obvious could you please change it to some readable var name?

Reply:
An artifact of it originally being a copy/paste job from `narrowTransformation`. Will change.